tsinfer/formats.py

```diff
@@ -1344,6 +1344,37 @@ def min_site_times(self, individuals_only=False):
             sites_bound = np.maximum(self.sites_time[:], sites_bound)
         return sites_bound
 
+    def modify_site_times(self, bespoke_times=None, individuals_only=False, **kwargs):
```

Comment:
I'm finding this method pretty unnecessary, to be honest. Isn't all we're doing, really,
```python
sites_time = # some value we've got somewhere; usually,
sites_time = self.min_site_times()
with sample_data.copy() as copy:
    copy.sites_time = sites_time

# Do something with copy?

Reply:
Yep, that's basically it. Happy to leave it out